src/createReducedStream.ts

```diff
@@ -0,0 +1,54 @@
+import { BehaviorSubject, Observable, Subscription, Subject } from 'rxjs';
+import { ActionStream } from './types/helpers';
+import { RegisteredReducer, combineReducers } from './reducer';
+import { defaultErrorSubject } from './internal/defaultErrorSubject';
+import { markName } from './internal/markers';
+import { tag } from 'rxjs-spy/operators';
+
+export class ReducedStream<State> extends BehaviorSubject<State> {
+  subscription: Subscription;
+
+  constructor(initialState: State, reducedState$: Observable<State>) {
+    super(initialState);
+
+    this.subscription = reducedState$.subscribe(this);
+  }
+}
+
+/**
+ * Create a reduced state stream
+ *
+ * A reduced state stream is a stream that scans over an action stream and other
+ * stream to build up a state. It is eternally subscribed and always exposes
+ * it latest value like a Behaviour subject.
+ *
+ * ```
+ * const myState$ = createReducedStream(
+ *   'myState$',
+ *   initialState,
+ *   reducers,
+ *   action$
+ * );
+ *
+ * myState$.value === initialState // Will be true
+ * ```
+ *
+ * @param name The name of the stream, used for placing a marker and spy tag
+ * @param initialState The initial state of the stream
+ * @param reducers The reducers that build up the stream state
+ * @param action$ The action stream the action reducers should reduce over
+ */
+export const createReducedStream = <State>(
+  name: string,
+  initialState: State,
+  reducers: RegisteredReducer<State, any>[],
+  action$: ActionStream,
+  errorSubject: Subject<any> = defaultErrorSubject
+): ReducedStream<State> => {
+  const reducedState$ = action$.pipe(
+    combineReducers(initialState, reducers, errorSubject),
+    markName(name),
+    tag(name)
```

Comment:
**This is copied from `reduceState`**, but does not contain `shareReplay` or `startWith`, because the `ReducedStream` handles that.

Do we still have a need for both? As of now I would say yes, because we need to use `reduceState` directly where we use namespaces, but once we have high level tooling for namespaces, I think the need will dissapear.

Reply:
I agree, would be nice to consolidate once we're able to :)